app/main.py

```diff
@@ -1 +1,32 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+        self,
+        name: str,
+        health: int = 100,
+        hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"""{{Name: {self.name}, """
+                f"""Health: {self.health}, Hidden: {self.hidden}}}""")
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, animal: Animal) -> None:
+        if animal.hidden is False and isinstance(animal, Herbivore):
```

Comment:
`if animal` or `if not animal` -> both statements return a boolean value, so no need to write `is False`